### PR TITLE
Agent: add option to emit CSV file for `evaluate-autocomplete`

### PR DESCRIPTION
**Pull request by olafurpg**
Previously, `evaluate-autocomplete` only generated snapshots to process the results. This PR adds a new CSV writer so that you can process/analyze the results with CSV tooling like xsv.

## Test plan

Ran large scale evaluation with the command
```
❯ pnpm agent evaluate-autocomplete --evaluation-config ~/dev/sourcegraph/bfg-evaluation/evaluation.json --test-count 400
```

### CODE REVIEW

File: agent/src/cli/evaluate-autocomplete/triggerAutocomplete.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    document: AutocompleteDocument
+
+    range: vscode.Range
+    modifiedContent: string
+    removedContent: string
+    position: vscode.Position
+    emptyMatchContent: string
+}
+
+export async function triggerAutocomplete(parameters: AutocompleteParameters): Promise<void> {
```
Comment (philipp-spiess) [inline agent/src/cli/evaluate-autocomplete/triggerAutocomplete.ts:21]: Nice cleanup

File: vscode/src/completions/artificial-delay.ts [hunk 2]
```diff
@@ -74,7 +74,9 @@ export function getArtificialDelay(
         userMetrics.currentLatency += featureFlags.user ? defaultLatencies.user : 0
     }
 
-    logDebug('CodyCompletionProvider:getLatency', `Delay added: ${total}`)
+    if (total > 0) {
+        logDebug('CodyCompletionProvider:getLatency', `Delay added: ${total}`)
+    }
```
Comment (philipp-spiess) [inline vscode/src/completions/artificial-delay.ts:79]: Let's rm this, we log this as part of the suggested event now anyways. I find this very annoying

File: agent/src/cli/evaluate-autocomplete/AutocompleteDocument.ts [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+     * @param diagnostic optional message to include with the formatted string
+     */
+    public format(range: vscode.Range, diagnostic?: string): string {
+        const line = this.lines[range.start.line]
+        const indent = ' '.repeat(range.start.character)
+        const length =
+            range.start.line === range.end.line
+                ? range.end.character - range.start.character
+                : line.length - range.start.character
+        const carets = length < 0 ? '<negative length>' : '^'.repeat(length)
```
Comment (philipp-spiess) [inline agent/src/cli/evaluate-autocomplete/AutocompleteDocument.ts:114]: BTW when using vscode.Range:

>	 * It is guaranteed that {@link Range.start start}.isBeforeOrEqual({@link Range.end end})

so negative length is not possible :)

**Review by philipp-spiess — APPROVED** [review]
Awesome!

File: agent/src/cli/evaluate-autocomplete/AutocompleteDocument.ts (hunk 3 above)
Comment (olafurpg) [reply agent/src/cli/evaluate-autocomplete/AutocompleteDocument.ts:114]: I don't think we enforce this in the shim!

File: vscode/src/completions/artificial-delay.ts (hunk 2 above)
Comment (olafurpg) [reply vscode/src/completions/artificial-delay.ts:79]: +1 to removing it. This diff here fixed my biggest pain

**Pull request merged**